src/main/java/co/cask/gcp/common/RecordFilterOutputFormat.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.gcp.common;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.RecordConverter;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.*;
+import org.apache.hadoop.mapreduce.lib.output.TextOutputFormat;
+import org.apache.orc.mapreduce.OrcOutputFormat;
+import parquet.avro.AvroParquetOutputFormat;
+
+import java.io.IOException;
+
+/**
+ * An OutputFormat that filters records before sending them to a delegate. Currently only supports TextOutputFormat
+ * as the delegate, but we may want to add other formats in the future.
+ */
+public class RecordFilterOutputFormat extends OutputFormat<NullWritable, StructuredRecord> {
```

Comment:
Also, use 2 space tab instead of 4. I think that's the standard used in CDAP. If you are using Intellij, there is a profile that's available here - https://wiki.cask.co/display/CE/Coding+Standard